x-pack/plugins/event_log/server/saved_object_provider_registry.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { KibanaRequest, SavedObjectsClientContract } from 'src/core/server';
+
+import { fromNullable, getOrElse } from 'fp-ts/lib/Option';
+import { pipe } from 'fp-ts/lib/pipeable';
+
+export type SavedObjectGetter = (
+  ...params: Parameters<SavedObjectsClientContract['get']>
+) => Promise<unknown>;
+export type SavedObjectProvider = (request: KibanaRequest) => SavedObjectGetter;
+
+export class SavedObjectProviderRegistry {
+  private providers = new Map<string, SavedObjectProvider>();
+  private defaultProvider?: SavedObjectProvider;
+
+  constructor() {}
+
+  public registerDefaultProvider(provider: SavedObjectProvider) {
+    this.defaultProvider = provider;
+  }
+
+  public registerProvider(type: string, provider: SavedObjectProvider) {
+    if (this.providers.has(type)) {
+      throw new Error(
+        i18n.translate(
```

Comment:
The i18n seems like overkill here - is there a case where `registerProvider()` would be called as part of an http request, vs run at setup/start - I'm guessing this message would only ever show up in the kibana log.

Reply:
Yeah, fair 👍 thu